haas/api.py

```diff
@@ -612,7 +708,8 @@ def port_detach_nic(port):
     """
     db = model.Session()
 
-    port = _must_find(db, model.Port, port)
+    switch = _must_find(db, model.Switch, switch)
+    port = _must_find_n(db, switch, model.Port, port)
 
```

Comment:
Added the check for a non-free node, and I've updated the `api-docs` branch (which there's a pr out for) with the new error accordingly.